Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py

```diff
@@ -3166,6 +3168,338 @@ def __init__(self, status_code) -> None:
+    ):
+        get_list_risky_users_or_host_command(client, command, {id: "test"})
+
+
+def test_get_list_user_groups_command(mocker):
+    """
+    Test function to validate the behavior of the `get_list_user_groups_command` function.
+
+    Args:
+        mocker: Pytest mocker object.
+        args (dict): A dictionary containing optional `group_names` argument.
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If the expected output doesn't match the actual output.
+    """
+    client = CoreClient("test", {})
+    test_data = load_test_data("./test_data/get_list_user_groups.json")
+    mocker.patch.object(CoreClient, "get_list_user_groups", return_value=test_data)
+
+    results = get_list_user_groups_command(client=client, args={"group_names": "test"})
+    assert test_data["reply"] == results.outputs
+
+
+@pytest.mark.parametrize(
+    "data, expected_results",
+    [
+        (
+            {
+                "group_name": "Group2",
+                "description": None,
+                "pretty_name": "dummy1",
+                "insert_time": 1111111111111,
+                "update_time": 2222222222222,
+                "user_email": [
+                    "dummy1@gmail.com",
+                    "dummy2@gmail.com",
+                ],
+                "source": "Custom",
+            },
+            [
+                {'User email': 'dummy1@gmail.com', 'Group Name': 'Group2', 'Group Description': None},
+                {'User email': 'dummy2@gmail.com', 'Group Name': 'Group2', 'Group Description': None}
+            ]
+        )
+    ],
+)
+def test_parse_user_groups(data: dict[str, Any], expected_results: list[dict[str, Any]]):
+    """
+    Test the 'parse_user_groups' function that parses user group information.
+
+    Args:
+        data (dict): A dictionary containing a sample user group data.
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If the parsing of user groups data fails.
+    """
+    results = parse_user_groups(data)
+
+    assert expected_results == results
+
+
+def test_get_list_user_groups_command_raise_exception(mocker):
+    """
+    Tests that the 'get_list_user_groups_command' function raises an exception when the 'get_list_user_groups' method of
+    the 'CoreClient' class raises a 'DemistoException'.
+
+    Args:
+        mocker: The pytest mocker object.
+
+    Raises:
+        Exception: If the 'get_list_user_groups_command' function does not raise an exception when expected.
+
+    Returns:
+        None.
+    """
+    client = CoreClient(
+        base_url="test",
+        headers={},
+    )
+
+    class MockException:
+        def __init__(self, status_code) -> None:
+            self.status_code = status_code
+
+    mocker.patch.object(
+        client,
+        "get_list_user_groups",
+        side_effect=DemistoException(
+            message="Group 'test' was not found", res=MockException(500)
```

Comment:
Do we need a `MockException`? Won't a `DemistoException` do? 🤔

Reply:
`'DemistoException' object has no attribute 'status_code'`

we need mock for the status_code

`if (
        e.res is not None
        and e.res.status_code == 500
        and 'was not found' in str(e)
    ):`